api/nodes/views.py

```diff
@@ -296,37 +295,23 @@ def bulk_destroy_skip_uneditable(self, resource_object_list, user, object_type):
 
     # Overrides BulkDestroyModelMixin
     def perform_bulk_destroy(self, resource_object_list):
-
-        auth = get_user_auth(self.request)
-        date = timezone.now()
-        id_list = [x.id for x in resource_object_list]
-
-        if NodeRelation.objects.filter(
-            parent__in=resource_object_list,
-            child__is_deleted=False,
-        ).exclude(Q(child__in=resource_object_list) | Q(is_node_link=True)).exists():
-            raise ValidationError('Any child components must be deleted prior to deleting this project.')
-
-        remove_addons(auth, resource_object_list)
+        if enforce_no_children(self.request):
+            if NodeRelation.objects.filter(
+                parent__in=resource_object_list,
+                child__is_deleted=False,
+            ).exclude(Q(child__in=resource_object_list) | Q(is_node_link=True)).exists():
+                raise ValidationError('Any child components must be deleted prior to deleting this project.')
```

Comment:
For versions prior to v2.12, bulk deleting children will still not be allowed if child components exist.  Legacy UI is still making requests to this endpoint.  For 2.12 and greater, deleting a node deletes every node in its hierarchy.

After 2.12, bulk deleting nodes should be discouraged.  Since deleting a node will delete every node in the hierarchy, this gets rid of most of our use cases for bulk delete anyway.  The bulk delete operation has the potential to be very expensive now, for example, sending in a request to delete ten nodes, each of which has a complex hierarchy that also needs to be deleted.